src/libs/actions/Report.js

```diff
@@ -49,6 +49,9 @@ const typingWatchTimers = {};
 // Keeps track of the max sequence number for each report
 const reportMaxSequenceNumbers = {};
 
+// Keeps track of the max sequence number for each report excluding loading actions
+const reportMaxSequenceNumbersNotLoading = {};
```

Comment:
I think we do include them, right? It gets updated [here](https://github.com/Expensify/Expensify.cash/blob/6960577fa798e491959a9abfc0e2e13d866a07ae/src/libs/actions/Report.js#L738), and addActions seems to be called immediately after a new comment is added

Reply:
Ok, so because we were previously "sourcing" things from the actions in the report view the `reportMaxSequenceNumbers` are not acceptable to use because some of them might be loading comments (in the new case, it looks like we are just adding +1 to whatever highest number we currently have and they are not GUID).

This all feels a little magical to me and like we could be handling the "loading" actions in a way that is easier to explain. I'm not sure I have a solution to off yet, but I do think adding another concept here makes things worse. It adds another layer of complexity and makes one have to do their own research to find out which flavor of "maxSequenceNumbers" they should be using.

Reply:
> adding another concept here makes things worse. It adds another layer of complexity and makes one have to do their own research to find out which flavor of "maxSequenceNumbers" they should be using.

I agree with that. I've played with `reportMaxSequenceNumbers` a bit.

I have tried breaking the `mark as unread` behavior (using `reportMaxSequenceNumbers`) and I haven't been able to. Since new comments from the user won't change `newMarkerSequenceNumber`, I think we are fine on that front.
And, for the other flow, worst case scenario, we'll set `sequenceNumber` to a sequence that will stop existing, right? To get the `unreadActionCount`, we do 
`const unreadActionCount = report.reportActionList.length - lastReadSequenceNumber;
  return Math.max(0, unreadActionCount);`
  
 So we are already handling `lastReadSequenceNumber` being too high (and thus not existing), so I think we are good there too!
 
 Thoughts?

Reply:
Sorry, I'm not quite following what you're saying here 😅 

Are you agreeing that we don't actually need `reportMaxSequenceNumbersNotLoading` and that we can change it.. or not?

Reply:
Lol.
Yes, I think we don't need `reportMaxSequenceNumbersNotLoading` and can just use `reportMaxSequenceNumbers`.